connector/server/explore/explore_query_logic.go

```diff
@@ -43,6 +43,66 @@ func NewExploreQuery(queryName string, query *models.ExploreQuery, user *models.
 	return new, new.isValid()
 }
 
+//PrepareAggregationValue Encrypts the subject count field of the SearchElement with the cothority key.
+func PrepareAggregationValue(subjectCountQueryInfo *models.ExploreSearchCountParams,
+	searchElement *models.ExploreSearchResultElement) (aggPatientFlags string, err error) {
+
+	if subjectCountQueryInfo == nil {
+		logrus.Debug("empty subject count query info")
+		return
+	}
+	userPublicKey := *subjectCountQueryInfo.UserPublicKey
+	baseQueryID := *subjectCountQueryInfo.QueryID
+
+	logrus.Debug("subject count is not empty ", searchElement.SubjectCount, " ", userPublicKey, " ", baseQueryID)
+	subjectCountInt, err := strconv.Atoi(searchElement.SubjectCount)
+	if err != nil {
+		logrus.Error("Cannot convert subject count to integer", err)
+		return
+	}
+
+	count64 := int64(subjectCountInt)
+
+	return unlynx.EncryptWithCothorityKey(count64)
+}
+
+//ExecuteTotalnumsAggregation executes an aggregate and key switch with other nodes of @param encryptedTotalnums
+func ExecuteTotalnumsAggregation(queryID string, encryptedTotalnums []string, userPublicKey string, queryType ExploreQueryType, timers medcomodels.Timers) (encAggCount []string, err error) {
+	var timer = time.Now()
+	var ksCountTimers map[string]time.Duration
+
+	if !queryType.Obfuscated {
```

Comment:
What is the desired behaviour when the query is obfuscated? At the moment this function will return an empty vector, and this will cause an error in the calling function. This should be managed better.